File: backend/src/main/java/com/revnu/backend/features/staff/dto/StaffRequest.java
package com.revnu.backend.features.staff.dto;

import java.math.BigDecimal;

import jakarta.validation.constraints.NotNull;
import jakarta.validation.constraints.Positive;

public record StaffRequest(
        String fullname,
        String position,
        @NotNull(message = "Salary rate is required")
        @Positive(message = "Salary rate must be greater than zero")
        BigDecimal salaryRate
        ) {

}

File: mobile/app/src/main/java/com/revnu/mobile/features/staff/model/StaffRequest.kt
package com.revnu.mobile.features.staff.model

import com.google.gson.annotations.SerializedName

data class StaffRequest(
    @SerializedName("fullname") val fullname: String,
    @SerializedName("position") val position: String,
    @SerializedName("salaryRate") val salaryRate: Double
)

